exporter/exporterhelper/factory_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterhelper
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+)
+
+const typeStr = "test"
+
+var defaultCfg = &configmodels.ExporterSettings{
+	TypeVal: typeStr,
+	NameVal: typeStr,
+}
+
+func TestNewTrace(t *testing.T) {
+	factory := NewFactory(
+		typeStr,
+		defaultConfig)
+	assert.EqualValues(t, typeStr, factory.Type())
+	assert.EqualValues(t, defaultCfg, factory.CreateDefaultConfig())
+	_, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, defaultCfg)
+	assert.Error(t, err)
+	_, err = factory.CreateMetricsExporter(context.Background(), component.ExporterCreateParams{}, defaultCfg)
+	assert.Error(t, err)
+
+	lfactory := factory.(component.LogExporterFactory)
+	_, err = lfactory.CreateLogExporter(context.Background(), component.ExporterCreateParams{}, defaultCfg)
+	assert.Error(t, err)
+}
+
+func TestNewMetrics_WithConstructors(t *testing.T) {
+	factory := NewFactory(
+		typeStr,
+		defaultConfig,
+		WithTraces(createTraceExporter),
+		WithMetrics(createMetricsExporter),
+		WithLogs(createLogExporter))
+	assert.EqualValues(t, typeStr, factory.Type())
+	assert.EqualValues(t, defaultCfg, factory.CreateDefaultConfig())
+
+	_, err := factory.CreateTraceExporter(context.Background(), component.ExporterCreateParams{}, defaultCfg)
+	assert.NoError(t, err)
```

Comment:
Should also validate that the trace exporter returned by `createTraceExporter` is returned here (similar for others)

Reply:
done.